pra-07/src/views/UserRegister.vue

```diff
@@ -0,0 +1,220 @@
+              <b-form-radio-group
+                v-model="form.selectedgender"
+                :options="form.genderOptions"
+                :aria-describedby="ariaDescribedby"
+                inline="true"
+              ></b-form-radio-group>
+              <b-form-invalid-feedback>
+                {{ errors[0] }}
+              </b-form-invalid-feedback>
+            </b-form-group>
+          </ValidationProvider>
+          <ValidationProvider
+            rules="required|numeric"
+            name="age"
+            v-slot="{ errors }"
+          >
+            <b-form-group label="age" class="mb-2">
+              <b-form-input
+                type="text"
+                v-model="form.age"
+                :state="errors[0] ? false : valid ? true : null"
+                placeholder="Enter age"
+                class="w-50"
+              >
+              </b-form-input>
+              <b-form-invalid-feedback>
+                {{ errors[0] }}
+              </b-form-invalid-feedback>
+            </b-form-group>
+          </ValidationProvider>
+          <ValidationProvider rules="required" name="dob" v-slot="{ errors }">
+            <b-form-group label="Choose Date Of Birth" class="mb-2 w-50">
+              <b-form-datepicker
+                v-model="form.dob"
+                class="mb-2"
+              ></b-form-datepicker>
+              <b-form-invalid-feedback>
+                {{ errors[0] }}
+              </b-form-invalid-feedback>
+            </b-form-group>
+          </ValidationProvider>
+          <b-button
+            type="submit"
+            pill
+            style="background-color: #1f2a48"
+            variant="outline-light"
+            >Register</b-button
+          >
+        </center>
+      </b-form>
+    </ValidationObserver>
+  </div>
+</template>
+
+<script>
+import AlertBox from "../components/AlertBox.vue";
+export default {
+  components: { AlertBox },
+  data() {
+    return {
+      form: {
+        name: "",
+        email: "",
+        password: "",
+        ucPassword: "",
+        age: "",
+        dob: "",
+        selectedrole: "null",
+        roleoptions: [
+          { value: null, text: "Select Role" },
+          { value: "Customer", text: "Customer" },
+          { value: "employee", text: "Employee" },
+          { value: "admin", text: "Admin" },
+        ],
+        selectedgender: "male",
+        genderOptions: [
+          { value: "male", text: "Male" },
+          { value: "female", text: "Female" },
+        ],
+      },
+      isLoading: false,
+      errmsg: "",
+      successmsg: "",
+    };
+  },
+  methods: {
+    async onSubmit(event) {
+      event.preventDefault();
+      this.userRegister(this.form);
+    },
+    async userRegister(form) {
+      let res = await this.$store.dispatch("auth/registerUser", form);
+      if (res === "success") {
+        this.$router.push({ name: "home" });
+      } else {
+        this.errmsg = "oops! somthing went wrong";
```

Comment:
in this.errormsg you can directly assign res because you have return error message in your registerUser action.